### ISSUE TITLE
Local builds with docker toolbox (Windows) fail if cmd argument is too long.

### ISSUE THREAD

**Issue by DrBomb** [cesanta/mos-tool#23]
Of course. I don't expect Windows to work as a build enviroment, but today I decided to experiment setting up docker toolbox on my machine just to compare build speed against a 4 CPU allocated Ubuntu VM.

After setting up and compiling some simple apps (5 libs) successfully, I proceeded to try to compile one of our "production" apps, and the thing fails with the message 

`Error: fork/exec docker.exe: The filename or extension is too long.`

Which then by googling came to this issue:

https://github.com/olikraus/U8glib_Arduino/issues/9#issuecomment-331202107

So basically local compilation on windows is doomed to fail when a lot of libs are added.

I don't know how does the latest Docker for Windows handles it though. Because it uses HyperV and I'm running a Home version of Windows.

**Comment by rojer** [comment]
yeah, we should start passing make variables as files.
